### PR TITLE
Pulls code for identifying which viewing key to use into separate utils

### PR DESCRIPTION
**Pull request by joeldudley**
### Why is this change needed?

In the future, the wallet extension will support multiple viewing keys. This will require the ability to ascertain which viewing key should be used for a given request. This can be established based on the contents of the request, but the logic to do so is currently spread out across the enclave code.

This PR pulls that code out so it can be reused cleanly.

### What changes were made as part of this PR:

Refactoring.

- Pulls code for parsing RPC requests, including establishing which viewing key to use, into a separate utils class
- Moves any non-encryption code out of RPCEncryptionManager, and into these new utils

### CODE REVIEW

File: go/enclave/rollupchain/rollup_chain.go [hunk 1]
```diff
@@ -667,57 +668,19 @@ func (rc *RollupChain) ExecuteOffChainTransaction(encryptedParams common.Encrypt
 	return encryptedResult, nil
 }
 
-// Extracts and validates the relevant parameters in a Call request.
-func extractCallParams(decryptedParams []byte) (gethcommon.Address, gethcommon.Address, []byte, error) {
```
Comment (joeldudley) [inline go/enclave/rollupchain/rollup_chain.go:671]: This was moved across into the RPC utils file, and broken into three separate methods.

File: go/enclave/rpc/rpc_encryption_manager.go [hunk 2]
```diff
@@ -100,48 +95,3 @@ func (rpc *RPCEncryptionManager) EncryptWithViewingKey(address gethcommon.Addres
 
 	return encryptedBytes, nil
 }
-
-// ExtractTxHash - Returns the transaction hash from a common.EncryptedParamsGetTxReceipt object.
-func (rpc *RPCEncryptionManager) ExtractTxHash(encryptedParams common.EncryptedParamsGetTxReceipt) (gethcommon.Hash, error) {
```
Comment (joeldudley) [inline go/enclave/rpc/rpc_encryption_manager.go:105]: Stuff like this isn't strictly encryption related, and has been moved off into the utils class.

File: go/enclave/rpc/rpc_utils.go [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+	if err != nil {
+		return gethcommon.Address{}, fmt.Errorf("could not parse JSON params in eth_getBalance request. JSON "+
+			"params are: %s. Cause: %w", string(balanceParams), err)
+	}
+	// The first argument is the address, the second the block.
+	return gethcommon.HexToAddress(paramsJSONMap[0]), nil
+}
+
+// ExtractCallParamTo extracts and parses the `to` field of an eth_call request.
+func ExtractCallParamTo(callParams []byte) (gethcommon.Address, error) {
```
Comment (BedrockSquirrel) [inline go/enclave/rpc/rpc_utils.go:76]: Could we have a CallParams struct and then just a single `ExtractCallParams` method that returns all of them bundled together. Would clear up the duplication here both in code and in CPU work,

**Review by BedrockSquirrel — COMMENTED** [review]
Generally LGTM, minor thought about the call data extraction

File: go/enclave/rpc/rpc_utils.go (hunk 3 above)
Comment (joeldudley) [reply go/enclave/rpc/rpc_utils.go:76]: I would do that, but on the WE side, we only need to extract one of them (the `from`), so I thought this was the best.

Comment (BedrockSquirrel) [reply go/enclave/rpc/rpc_utils.go:76]: Personally I'd rather extract the fields unnecessarily for the WE than duplicate the code and work in the enclave but don't feel strongly.

**Review by BedrockSquirrel — APPROVED** [review]
LGTM

**Pull request merged**